pkg/cli/cmds/root.go

```diff
@@ -9,12 +9,31 @@ import (
 
 	"github.com/k3s-io/k3s/pkg/version"
 	"github.com/rancher/rke2/pkg/images"
+	"github.com/rancher/rke2/pkg/podtemplate"
 	"github.com/rancher/rke2/pkg/rke2"
 	"github.com/urfave/cli/v2"
 )
 
 var (
-	appName    = filepath.Base(os.Args[0])
+	appName = filepath.Base(os.Args[0])
+	config  = rke2.Config{
+		ExtraEnv: map[string]*cli.StringSlice{
+			podtemplate.KubeAPIServer:          &cli.StringSlice{},
+			podtemplate.KubeScheduler:          &cli.StringSlice{},
+			podtemplate.KubeControllerManager:  &cli.StringSlice{},
+			podtemplate.KubeProxy:              &cli.StringSlice{},
+			podtemplate.Etcd:                   &cli.StringSlice{},
+			podtemplate.CloudControllerManager: &cli.StringSlice{},
+		},
+		ExtraMounts: map[string]*cli.StringSlice{
+			podtemplate.KubeAPIServer:          &cli.StringSlice{},
+			podtemplate.KubeScheduler:          &cli.StringSlice{},
+			podtemplate.KubeControllerManager:  &cli.StringSlice{},
+			podtemplate.KubeProxy:              &cli.StringSlice{},
+			podtemplate.Etcd:                   &cli.StringSlice{},
+			podtemplate.CloudControllerManager: &cli.StringSlice{},
+		},
+	}
```

Comment:
I prefer the old way of predefined struct of StringSlices. We are never utilizing the dynamic nature of Maps, no new additions or deletions, so all this empty assignment and then reassignment with `parseControlPlaneMounts` is more complex and confusing over what we had before. 

Just keep the ExtraMounts and ExtraEnvs.